File: 2025-1/sesion-de-clase-17/SoftProgNet/SoftProgNegocio/BOImpl/ProductoBOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.Logistica.Almacen;
using PUCP.Edu.Pe.SoftProg.Negocio.BO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.Logistica.Almacen;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Logistica.Almacen;

namespace PUCP.Edu.Pe.SoftProg.Negocio.BOImpl {
    public class ProductoBOImpl : IProductoBO {
        private readonly IProductoDAO productoDAO;

        public ProductoBOImpl() {
            this.productoDAO = new ProductoDAOImpl();
        }

        public void Guardar(Producto producto, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.productoDAO.Insertar(producto);
            }
            else { 
                this.productoDAO.Modificar(producto);
            }
        }

        public void Eliminar(int id) {
            this.productoDAO.Eliminar(id);
        }

        public Producto Obtener(int id) {
            return this.productoDAO.Buscar(id);
        }

        public List<Producto> Listar() {
            return this.productoDAO.Listar();
        }        
    }
}

File: 2025-1/sesion-de-clase-20/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/ProductoBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.IProductoBO;
import pe.edu.pucp.inf30.softprog.dao.logistica.almacen.IProductoDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.logistica.almacen.ProductoDAOImpl;
import pe.edu.pucp.inf30.softprog.model.Estado;
import pe.edu.pucp.inf30.softprog.model.logistica.almacen.Producto;

/**
 *
 * @author eric
 */
public class ProductoBOImpl implements IProductoBO {
    private final IProductoDAO productoDao;

    public ProductoBOImpl() {
        this.productoDao = new ProductoDAOImpl();
    }
    
    @Override
    public List<Producto> listar() {
        return this.productoDao.listar();
    }

    @Override
    public Producto obtener(int id) {
        return this.productoDao.buscar(id);
    }

    @Override
    public void eliminar(int id) {
        this.productoDao.eliminar(id);
    }

    @Override
    public void guardar(Producto modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.productoDao.insertar(modelo);
        }
        else {
            this.productoDao.modificar(modelo);
        }
    }
}
